src/web/rustdoc.rs

```diff
@@ -238,26 +239,28 @@ pub fn rustdoc_html_server_handler(req: &mut Request) -> IronResult<Response> {
         return Ok(super::redirect(canonical));
     }
 
-    let path = {
-        let mut path = req_path.join("/");
-        if path.ends_with('/') {
-            req_path.pop(); // get rid of empty string
-            path.push_str("index.html");
-            req_path.push("index.html");
-        }
-        path
+    let mut path = req_path.join("/");
+
+    //If it is a directory, we default to looking 
+    if path.ends_with("/") {
+        path.push_str(DIR_DEFAULT_FILE);
+        req_path.push(DIR_DEFAULT_FILE);
```

Comment:
I don't think this is the same behavior, it will turn `/crate/version/` into `/crate/version//index.html`.

This will mess up the redirect to latest version, which unfortunately only has unit tests in #570.

Since that's the case, I'm not comfortable merging until either #570 is merged or I get a chance to test locally which will probably be a few days.

If you instead change _only_ the `from_path` block (line 251), I would be more comfortable with the changes.